src/python_bindings/bindings.cpp

```diff
@@ -121,17 +121,17 @@ class PyHNSWLibIndex : public PyVecSimIndex {
     }
 
     void setDefaultEf(size_t ef) {
-        auto *hnsw = reinterpret_cast<HNSWIndex *>(index);
+        auto *hnsw = reinterpret_cast<HNSWIndex<float, float> *>(index);
```

Comment:
Are we going to template the bindings as well?

Reply:
in my opinion only should be templated PyHNSWLibIndex

Reply:
What is the proposed solution for additional types here?